src/Contracts/OpdexMiningGovernance.cs

```diff
@@ -191,21 +188,19 @@ public void RewardMiningPool()
         
         var nominations = Nominations;
         
-        for (uint i = 0; i < MaxNominations; i++)
+        for (uint i = 0; i < nominations.Length; i++)
         {
-            if (nominations[i].StakingPool == Address.Zero) continue;
+            if (nominations[i].Weight == UInt256.Zero) continue;
             
             RewardMiningPoolExecute(nominations[i], MiningPoolReward);
             IncrementMiningPoolsFunded();
+
+            nominations[i].Weight = UInt256.Zero;
+            Nominations = nominations;
```

Comment:
Same problematic scenario as mentioned above. Maybe a new Flag of some sort on each Nominations. 

Need a way to say "This pool has been funded for **this** round already so skip", then next round if nothing's changed, be able to fund again.